typed-racket-test/succeed/prefab.rkt

```diff
@@ -53,3 +53,45 @@
 ;; Test match (indirectly tests unsafe-struct-ref)
 (match (foo 'x) [(foo s) s])
 (match (foo* 'x) [(foo* s) s])
+
+
+
+(struct msg ([str : String] [maybe-str : (Option String)]) #:prefab)
+
+;; tests let bindings fields of prefab structs
+;; works as expected (requires path-type.rkt to be correct w.r.t. prefabs)
+(: test-let-binding (-> Any String))
+(define (test-let-binding x)
+  (cond
+    [(msg? x)
```

Comment:
Yes. Immutable fields in sane prefab structs would certainly be things we wanted paths to be able to reference (and I believe we're already currently ommitting mutable fields), and the additions this PR makes (making update/path-type able to reach into prefab types) certainly would make sense in that setting.

My only concern is more programs being created after this PR that would then be broken by fixing prefabs... but I'm not positive this would be an issue, it's just a kneejerk concern at the moment...